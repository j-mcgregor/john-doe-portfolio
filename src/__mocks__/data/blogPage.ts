import _ from 'lodash'
import {
    BlogPageProps,
    PRISMIC_PRIMARY_Image_with_Caption,
    PRISMIC_PRIMARY_Quote,
    PRISMIC_PRIMARY_Text,
} from '../../types/interfaces/blog'
import {
    PRISMIC_blog_node,
    PRISMIC_RichTextType,
    PRISMIC_MetaType,
} from '../../types/interfaces/prismic'

type ImageType = 'image_with_caption' | 'full-width-image' | 'emphasized'

type TextType = 'paragraph' | 'heading3'

export interface ImageSliceTypeProps {
    [index: string]: ImageType
}

export const imageSliceType: ImageSliceTypeProps = {
    '0': 'image_with_caption',
    '1': 'full-width-image',
    '2': 'emphasized',
}

export const makeTextType = (
    type: TextType,
    text: string
): PRISMIC_RichTextType => ({
    type,
    text,
    spans: [],
})

export const meta = (
    id: string,
    uid: string,
    type: string
): PRISMIC_MetaType => ({
    id,
    uid,
    type,
})

export const imageSlice = (
    imageType: number
): PRISMIC_PRIMARY_Image_with_Caption => ({
    primary: {
        caption: [makeTextType('paragraph', `caption-${imageType}`)],
        image: {
            dimensions: {
                width: 1200,
                height: 900,
            },
            alt: `alt-${imageType}`,
            copyright: null,
            url: 'http://fillmurray.com/300/300',
        },
    },
    label: `default`,
    type: imageSliceType[`${imageType}`],
})

export const quoteSlice = (text: string): PRISMIC_PRIMARY_Quote => ({
    primary: {
        quote: [makeTextType('paragraph', text)],
    },
    label: undefined,
    type: 'quote',
})

export const textSlice = (text: string): PRISMIC_PRIMARY_Text => ({
    primary: {
        text: [makeTextType('paragraph', text)],
    },
    label: undefined,
    type: 'text',
})

export const blogNodes: PRISMIC_blog_node[] = _.times(3, num => ({
    node: {
        date: `2020-0${num + 1}-01`,
        body: [
            textSlice('Lorem ipsum dolor sit amet'),
            imageSlice(num),
            quoteSlice('And I quoteth thus...'),
        ],
        _meta: meta(`id-${num + 1}`, `blog-${num + 1}`, 'blog_post'),
        title: [makeTextType('paragraph', `Post ${num + 1}`)],
    },
}))

export const blogPage: BlogPageProps = {
    data: {
        prismic: {
            allBlog_posts: {
                edges: blogNodes,
            },
        },
    },
}
