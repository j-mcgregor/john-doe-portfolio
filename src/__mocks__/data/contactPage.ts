export const contactPage = {
    data: {
        prismic: {
            allContacts: {
                edges: [
                    {
                        node: {
                            title: [
                                {
                                    type: 'heading3',
                                    text: 'Contact Us',
                                    spans: [],
                                },
                            ],
                            subtitle: [
                                {
                                    type: 'heading4',
                                    text: 'Leave a message',
                                    spans: [],
                                },
                            ],
                            social_links: [],
                        },
                    },
                ],
            },
        },
    },
}
