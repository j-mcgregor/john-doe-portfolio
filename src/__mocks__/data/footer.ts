export const footerData = {
    prismic: {
        allContacts: {
            edges: [
                {
                    node: {
                        social_links: [
                            {
                                name: 'Mail',
                                text: 'info@johndoe.com',
                                url: {
                                    url: 'mailto:info@johndoe.com',
                                },
                            },
                            {
                                name: 'Phone',
                                text: '+12345678900',
                                url: {
                                    url: 'tel:+12345678900',
                                },
                            },
                            {
                                name: 'LinkedIn',
                                text: null,
                                url: {
                                    url: 'https://linkedin.com',
                                },
                            },
                            {
                                name: 'Facebook',
                                text: null,
                                url: {
                                    url: 'https://facebook.com',
                                },
                            },
                            {
                                name: 'Twitter',
                                text: null,
                                url: {
                                    url: 'https://twitter.com',
                                },
                            },
                            {
                                name: 'Address',
                                text: '123 Main Street, New York, USA',
                                url: null,
                            },
                        ],
                    },
                },
            ],
        },
    },
}
