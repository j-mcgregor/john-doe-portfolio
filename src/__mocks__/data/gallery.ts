import { times } from 'lodash'

export const galleryPage = {
    data: {
        allFile: {
            edges: [
                {
                    node: {
                        name: 'grid-svg.inline',
                        id: 'grid-svg.inline',
                        publicURL: '/assets/grid-svg.inline.svg',
                    },
                },
                {
                    node: {
                        name: 'grid-col-svg.inline',
                        id: 'grid-col-svg.inline',
                        publicURL: '/assets/grid-col-svg.inline.svg',
                    },
                },
                {
                    node: {
                        name: 'grid-row-svg.inline',
                        id: 'grid-row-svg.inline',
                        publicURL: '/assets/grid-row-svg.inline.svg',
                    },
                },
            ],
        },
        prismic: {
            allGallerys: {
                edges: [
                    {
                        node: {
                            title: [
                                {
                                    type: 'heading3',
                                    text: 'Gallery',
                                    spans: [],
                                },
                            ],
                            description: [
                                {
                                    type: 'heading4',
                                    text: 'Look at my stuff!!',
                                    spans: [],
                                },
                            ],
                            body: [
                                {
                                    type: 'type',
                                    fields: times(10, num => ({
                                        alt_text: [
                                            {
                                                type: 'paragraph',
                                                text: `alt text ${num}`,
                                                spans: [],
                                            },
                                        ],
                                        caption: [
                                            {
                                                type: 'paragraph',
                                                text: `caption ${num}`,
                                                spans: [],
                                            },
                                        ],
                                        image: {
                                            dimensions: {
                                                width: 1000,
                                                height: 800,
                                            },
                                            alt: `alt text ${num}`,
                                            copyright: `copyright ${num}`,
                                            url: `http://something-${num}.com`,
                                        },
                                    })),
                                },
                            ],
                        },
                    },
                ],
            },
            allContacts: {
                edges: [
                    {
                        node: {
                            social_links: [
                                {
                                    name: 'Mail',
                                    text: 'info@johndoe.com',
                                    url: {
                                        url: 'mailto:info@johndoe.com',
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    },
}

export const localFiles = {
    allFile: {
        edges: [
            {
                node: {
                    name: 'grid-svg.inline',
                    id: 'grid-svg.inline',
                    publicURL: '/assets/grid-svg.inline.svg',
                },
            },
            {
                node: {
                    name: 'grid-col-svg.inline',
                    id: 'grid-col-svg.inline',
                    publicURL: '/assets/grid-col-svg.inline.svg',
                },
            },
            {
                node: {
                    name: 'grid-row-svg.inline',
                    id: 'grid-row-svg.inline',
                    publicURL: '/assets/grid-row-svg.inline.svg',
                },
            },
        ],
    },
}

export const imageList = times(20, num => ({
    alt_text: [{ text: 'text', type: 'type', spans: [] }],
    image: {
        dimensions: {
            width: 300,
            height: 300,
        },
        alt: `test-${num}`,
        copyright: `test-${num}`,
        url: `test-${num}`,
        thumbnail: {
            dimensions: {
                width: 300,
                height: 300,
            },
            alt: `test-${num}`,
            copyright: `test-${num}`,
            url: `test-${num}`,
        },
    },
    caption: [{ text: 'text', type: 'type', spans: [] }],
}))
