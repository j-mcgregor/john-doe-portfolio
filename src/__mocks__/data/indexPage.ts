export const indexPage = {
    prismic: {
        allLandings: {
            edges: [
                {
                    node: {
                        primary_text: [
                            {
                                type: 'header1',
                                text: 'John Doe',
                                spans: [],
                            },
                        ],
                        secondary_text: [
                            {
                                type: 'header2',
                                text: 'Developer',
                                spans: [],
                            },
                        ],
                        background_image: {
                            copyright: '',
                            dimensions: {
                                height: 100,
                                width: 100,
                            },
                            url: 'logo.png',
                            alt: 'logo',
                        },
                        about: [
                            {
                                type: 'paragraph',
                                text: 'lorem ipsum',
                                spans: [],
                            },
                        ],
                        experience: [
                            {
                                city: 'London',
                                company: 'London Inc',
                                date_from: '2019-01-01',
                                date_to: '2020-01-01',
                                description: [
                                    {
                                        type: 'paragraph',
                                        text: 'some job description here',
                                        spans: [],
                                    },
                                ],
                                job_title: 'Dev',
                            },
                        ],
                    },
                },
            ],
        },
    },
    file: {
        childImageSharp: {
            fixed: {
                height: 200,
                width: 200,
                base64: '/logo.png',
                src: '/logo.png',
                srcSet: '/logo.png',
            },
        },
    },
}
