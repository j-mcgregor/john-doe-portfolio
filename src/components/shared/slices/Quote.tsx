import React from 'react'
import RichText from '../../../utils/RichTextCustom'

export const Quote = ({ slice }) => (
    <div className="post-quote container">
        <blockquote>{<RichText render={slice.primary.quote} />}</blockquote>
    </div>
)
