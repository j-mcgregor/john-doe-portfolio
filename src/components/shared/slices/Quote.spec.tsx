import React from 'react'
import { render } from '@testing-library/react'
import { Quote } from './Quote'
import * as data from '../../../__mocks__/data/blogPage'

describe('Quote', () => {
    it('should render', () => {
        const props = { slice: data.quoteSlice('Join the Dark Side') }

        const { baseElement } = render(<Quote {...props} />)

        expect(baseElement).toMatchSnapshot()
    })
})
