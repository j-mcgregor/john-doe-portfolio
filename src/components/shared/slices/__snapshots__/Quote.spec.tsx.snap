// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render 1`] = `
<body>
  <div>
    <div
      class="post-quote container"
    >
      <blockquote>
        <p>
          Join the Dark Side
        </p>
      </blockquote>
    </div>
  </div>
</body>
`;
